software/control/microscope.py

```diff
@@ -179,4 +180,149 @@ def close(self):
         self.camera.close()
         self.microcontroller.close()
         if USE_ZABER_EMISSION_FILTER_WHEEL or USE_OPTOSPIN_EMISSION_FILTER_WHEEL:
-            self.emission_filter_wheel.close()
\ No newline at end of file
+            self.emission_filter_wheel.close()
+
+
+
+class LightSourceType(Enum):
+    Squid = 0
+    LDI = 1
+    CELESTA = 2
+    VersaLase = 3
+    SCI = 4
+
+class IntensityControlMode(Enum):
+    OnBoardDAC = 0
+    Software = 1
+    USB = 2
+
+class ShutterControlMode(Enum):
+    TTL = 0
+    Software = 1
+    USB = 2
+
+class IlluminationController():
+    def __init__(self, microcontroller, intensity_control, shutter_control, light_source_type=None, light_source=None):
+        self.microcontroller = microcontroller
+        self.intensity_control_mode = intensity_control
+        self.shutter_control_mode = shutter_control
+        self.light_source_type = light_source_type
+        self.light_source = light_source
+        self.channel_mappings_TTL = {
+            405: 11,
+            470: 12,
+            488: 12,
+            545: 14,
+            550: 14
+            555: 14,
+            561: 14,
+            638: 13,
+            640: 13
+            730: 15,
+            735: 15,
+            750: 15
+        }
+        
+        self.channel_mappings_software = {}
+        self.is_on = {}
+        self.intensity_settings = {}
+        self.pmin, self.pmax = 0, 1000
+        self.current_channel = None
+
+        if self.light_source_type is not None:
+            self.configure_light_source()
+
+    def configure_light_source(self):
+        if self.light_source_type == LightSourceType.CELESTA:
+            self.intensity_control_mode = IntensityControlMode.Software
+            self.set_shutter_control_mode(ShutterControlMode.TTL)
+            self.shutter_control_mode = ShutterControlMode.TTL
```

Comment:
We should use `self.intensity_control_mode` and `self.set_shutter_control_mode`. Specification of which mode to use should be in the config file.